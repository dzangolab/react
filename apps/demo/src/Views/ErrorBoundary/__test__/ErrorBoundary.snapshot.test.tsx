import { render } from "@testing-library/react";
import { expect, test } from "vitest";

import ErrorBoundary from "../index";

vi.mock("react-router-dom", async () => ({
  ...((await vi.importActual("react-router-dom")) as any),
  useRouteError: () => ({ statusText: 404, message: "Not Found" }),
}));

test("Component matches snapshot", async () => {
  const { container } = render(<ErrorBoundary />);

  expect(container).toMatchSnapshot();
});
