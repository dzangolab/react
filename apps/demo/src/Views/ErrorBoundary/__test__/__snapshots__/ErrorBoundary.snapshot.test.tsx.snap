// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div>
  <div
    class="dz-page"
    data-centered="false"
  >
    <div
      class="dz-page-header"
    >
      <div
        class="dz-page-title-wrapper"
      />
      <div
        class="dz-page-toolbar"
        data-testid="page-toolbar"
      />
    </div>
    <div
      class="dz-page-content"
      data-testid="page-content"
    >
      <div
        class="error-page"
      >
        <h1>
          errorBoundary.title
        </h1>
        <p>
          errorBoundary.subtitle
        </p>
        <p>
          <i>
            404
          </i>
        </p>
      </div>
    </div>
  </div>
</div>
`;
